src/RequestExecutor/HttpRequestExecutor.spec.ts

```diff
@@ -142,5 +142,20 @@ describe('HttpRequestExecutor', () => {
 
       (typeof response.statusCode).should.equal('undefined');
     });
+
+    it('do not truncate whitelisted mime types', async () => {
+      const { request, requestOptions } = createRequest();
+      const big_body = 'I will always cover my code with unit tests'.repeat(
```

Comment:
lets without easter eggs 😄 

```suggestion
      const bigBody = 'Too big body'.repeat(
```